### ISSUE TITLE
Problem running Rocket.Chat

### ISSUE THREAD

**Issue by Syirrus** [RocketChat/Rocket.Chat#5088]
Your Rocket.Chat version: (make sure you are running the latest)

When I updated Rocket.Chat and run: meteor, I get the following in the rocket.chat log:

W20161201-14:09:44.154(-5)? (STDERR) ufs: permissions are not defined for store "fileSystem"
I20161201-14:09:44.156(-5)? ufs: store created at

**Comment by forgulencia** [comment]
I have exactly the same problem with a fresh installation, on a new server (Ubuntu 14.04) followed the docs from here:

https://rocket.chat/docs/installation/manual-installation/ubuntu

Here is what I see after exporting the following:

```
export ROOT_URL='http://rocket.example.com:4000'
export MONGO_URL='mongodb://localhost:27017/rocketchat'
export PORT='4000'
export MAIL_URL='smtp://XXXXXXXXXX:YYYYYYYYYY/ZZZZZZZZZZZZ@email-smtp.us-east-1.amazonaws.com:587/'
```
Tried with or without parenthesis, removed mail_url completely, resintalled nodejs npm, everything. 

Updating process.env.MAIL_URL
ufs: permissions are not defined for store "fileSystem"
ufs: store created at 
{"line":"159","file":"rocketchat_migrations.js","message":"Migrations: Not migrating, already at version 64","time":{"$date":1480671328688},"level":"info"}
Updating process.env.MAIL_URL

And it stucks here forever. If it helps I can see the collections generated in the mongo database. Nothing occupies port 4000 (also tried with 3000 w/o luck). No error msg, nothing. 

Can somebody provide help? Thanks

**Comment by hugops** [comment]
I have the same problem, after creating a new database mongo (Changing the mongo path) returned to work.

**Comment by Syirrus** [comment]
Has anyone found a workaround for this? I'm still dead in the water.

**Comment by Sing-Li** [comment]
Judging from our recent experience in #support on https://demo.rocket.chat - which , by the way, is probably the most effective way to tap into the experience of our 90,000 +  users.

It is MOST LIKELY due to 1 of  2 things:

1)  some database muck-up , so starting from a clean database fixes it

2)  you are using an OLD VERSION of Mongo that is not supported;  check the version - default from Ubuntu 14.04 is too old -- you need absolutely 2.6.10 or higher

Just sharing what we've learnt so far.   Good luck.

**Comment by Syirrus** [comment]
@Sing-Li Thank you for chiming in. I removed my DB from my local machine and it didn't do anything. I still had the same problem. However, when I re-ran `meteor npm start` it loaded up. 
Hopefully that works for others who are facing this issue on their installs.

**Comment by Sing-Li** [comment]
@Syirrus - thanks for the follow up!

**Comment by forgulencia** [comment]
@Sing-Li: Thank you very much, the second case was my problem. It was funny because the collections were generated, and everything seemed ok. How about some dependency check before the server starts?

**Comment by th3m4ri0** [comment]
I have the same issue with docker : 
https://asciinema.org/a/5lae8p46a7nlqha2g23xxs7jb

```
root@rocket:~# docker run --name db -d mongo:3.0 --smallfiles                                                                                                 
0610e2b9d8be6a03224c50745f98f0567a5b0a47402cc136b2f2153f1ce3f740                                                                                              
root@rocket:~# docker run --name rocketchat -p 80:3000 --env ROOT_URL=http://localhost --link db -d rocket.chat                                               
6da8d036f4b6235c14452634b62fe047597b42ac3be7c2221ad738c712e7432c                                                                                              
root@rocket:~# docker ps                                                                                                                                      
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                  NAMES                         
6da8d036f4b6        rocket.chat         "node main.js"           6 seconds ago       Up 5 seconds        0.0.0.0:80->3000/tcp   rocketchat                    
0610e2b9d8be        mongo:3.0           "/entrypoint.sh --sma"   13 seconds ago      Up 12 seconds       27017/tcp              db                            
root@rocket:~# docker ps                                                                                                                                      
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                  NAMES                         
6da8d036f4b6        rocket.chat         "node main.js"           20 seconds ago      Up 19 seconds       0.0.0.0:80->3000/tcp   rocketchat                    
0610e2b9d8be        mongo:3.0           "/entrypoint.sh --sma"   27 seconds ago      Up 26 seconds       27017/tcp              db                            
root@rocket:~# docker ps                                                                                                                                      
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS               NAMES                            
0610e2b9d8be        mongo:3.0           "/entrypoint.sh --sma"   37 seconds ago      Up 37 seconds       27017/tcp           db                               
root@rocket:~# docker logs rocketchat                                                                                                                         
Updating process.env.MAIL_URL                                                                                                                                 
ufs: permissions are not defined for store "fileSystem"                                                                                                       
ufs: store created at                                                                                                                                         
root@rocket:~# docker ps -a                                                                                                                                   
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                        PORTS               NAMES                  
6da8d036f4b6        rocket.chat         "node main.js"           45 seconds ago      Exited (137) 14 seconds ago                       rocketchat             
0610e2b9d8be        mongo:3.0           "/entrypoint.sh --sma"   52 seconds ago      Up 51 seconds                 27017/tcp           db    
```

**Comment by baoailian** [comment]
I have the same issue with @th3m4ri0.

running in docker with centos6.7
```
[root@ec1e5326255a Rocket.Chat]# node main.js
Updating process.env.MAIL_URL
ufs: permissions are not defined for store "fileSystem"
ufs: store created at
ufs: permissions are not defined for store "fileSystem"
ufs: store created at
{"line":"160","file":"rocketchat_migrations.js","message":"Migrations: Not migrating, already at version 68","time":{"$date":1481303342621},"level":"info"}
Updating process.env.MAIL_URL
```

It doesn't work any more.

I try the same steps it in centos7 vm, it works ok! 

By the way, running directly  from your docker-compose.yml  also a mistake.

```
[root@sdfwe923 demo]# docker images
rocketchat/hubot-rocketchat   latest              4a259deaf083        6 weeks ago         800.3 MB
[root@sdfwe923 demo]#
[root@sdfwe923 demo]# docker ps -a
CONTAINER ID        IMAGE                                COMMAND                CREATED             STATUS                        PORTS                    NAMES
2afbdebf97ef        rocketchat/hubot-rocketchat:latest   "/bin/sh -c 'node -e   7 minutes ago       Up 7 minutes                  0.0.0.0:3001->8080/tcp   demo_hubot_1
8a6b6f9b3343        rocketchat/rocket.chat:latest        "node main.js"         8 minutes ago       Exited (1) 2 minutes ago                               demo_rocketchat_1
```

**Comment by th3m4ri0** [comment]
Debian 8 works fine too.

**Comment by rychken** [comment]
Sharing what just worked for me.

1) I launched with `meteor` from another directory / repo. That succeeded. Weird. Ok, something about my dir? I killed this instance.
2) I renamed the problematic directory / repo.
3) I launched from the renamed directory. That succeeded.

Before this I tried everything. I verified versions. I reset meteor, cleared mongo. I `git clean -ffdx`'d. I rebooted. I tried various previous commits that I know for sure worked. I even wiped `~/.meteor`.

4) I renamed my directory BACK to what it was. Relaunched. Succeeded.
